`syrupy` testing fails without `pydantic` installed

**Describe the bug**

When I run tests for version 5.3.0 (without `pydantic` installed), then I do see this failure:
```
==================================== ERRORS ====================================
_ ERROR collecting tests/syrupy/extensions/amber/test_amber_serializer_plugins.py _
ImportError while importing test module '/data/builds/ul-workspace/components/python/syrupy/build/amd64-3.14/tests/syrupy/extensions/amber/test_amber_serializer_plugins.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.14/vendor-packages/_pytest/python.py:507: in importtestmodule
    mod = import_path(
/usr/lib/python3.14/vendor-packages/_pytest/pathlib.py:587: in import_path
    importlib.import_module(module_name)
/usr/lib/python3.14/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
<frozen importlib._bootstrap>:1406: in _gcd_import
    ???
<frozen importlib._bootstrap>:1371: in _find_and_load
    ???
<frozen importlib._bootstrap>:1342: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:938: in _load_unlocked
    ???
/usr/lib/python3.14/vendor-packages/_pytest/assertion/rewrite.py:197: in exec_module
    exec(co, module.__dict__)
tests/syrupy/extensions/amber/test_amber_serializer_plugins.py:7: in <module>
    from pydantic import BaseModel
E   ModuleNotFoundError: No module named 'pydantic'
_ ERROR collecting tests/syrupy/extensions/amber/test_amber_serializer_plugins.py _
ImportError while importing test module '/data/builds/ul-workspace/components/python/syrupy/build/amd64-3.14/tests/syrupy/extensions/amber/test_amber_serializer_plugins.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.14/vendor-packages/_pytest/python.py:507: in importtestmodule
    mod = import_path(
/usr/lib/python3.14/vendor-packages/_pytest/pathlib.py:587: in import_path
    importlib.import_module(module_name)
/usr/lib/python3.14/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
<frozen importlib._bootstrap>:1406: in _gcd_import
    ???
<frozen importlib._bootstrap>:1371: in _find_and_load
    ???
<frozen importlib._bootstrap>:1342: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:938: in _load_unlocked
    ???
/usr/lib/python3.14/vendor-packages/_pytest/assertion/rewrite.py:197: in exec_module
    exec(co, module.__dict__)
tests/syrupy/extensions/amber/test_amber_serializer_plugins.py:7: in <module>
    from pydantic import BaseModel
E   ModuleNotFoundError: No module named 'pydantic'
=========================== short test summary info ============================
ERROR tests/syrupy/extensions/amber/test_amber_serializer_plugins.py
ERROR tests/syrupy/extensions/amber/test_amber_serializer_plugins.py
!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
```

**To reproduce**

Run tests for `syrupy` without `pydantic` installed.

**Expected behavior**

Either no test failure, or `pydantic` is in the `test` dependency group.

**Screenshots**

NA

**Environment (please complete the following information):**

- OS: OpenIndiana
- Syrupy Version: 5.3.0
- Python Version: 3.14.4

**Additional context**

NA

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -32,17 +32,18 @@ syrupy = "syrupy"
 test = [
     "invoke>=2.2.0,<3",
     "coverage[toml]>=7.6.12,<8",
-    "hypothesis>=6.0.0",
+    "hypothesis>=6.0.0,<7",
     "pytest-benchmark>=5.1.0,<6",
     "pytest-xdist>=3.6.1,<4",
+    "pydantic>=2.13.4,<3",
+    "attrs>=26.1.0,<27",
 ]
 dev = [
     "mypy>=2.1.0,<3",
     "py-githooks>=1.1.1,<2",
     "debugpy>=1.8.20,<2",
-    "ruff>=0.15.15",
-    "pydantic>=2.13.4",
-    "attrs>=26.1.0"
+    "ruff>=0.15.15,<0.16",
+    "attrs>=26.1.0,<27",
 ]
 
 [tool.uv]

